sklearn/metrics/ranking.py

```diff
@@ -100,7 +100,11 @@ def auc(x, y, reorder=False):
                                  "the x array is not increasing: %s" % x)
 
     area = direction * np.trapz(y, x)
-
+    if isinstance(area, np.memmap):
+        # Reductions such as .sum used internally in np.trapz do not return a
+        # scalar float by default for numpy.memmap instances contrary to
+        # traditional arrays.
+        area = float(area)
```

Comment:
We don't we always convert to float? Right now we'll have numpy floats or floats depending on the parallel computing setting, which might surprise people (different set of methods on the different objects).